scarb/src/ops/resolve.rs

```diff
@@ -67,6 +69,15 @@ pub fn generate_compilation_units(
 ) -> Result<Vec<CompilationUnit>> {
     let mut units = Vec::with_capacity(ws.members().size_hint().0);
     for member in ws.members() {
+        let current_version = Version::parse(&*get().cairo.version)?;
+        // Remove any pre-release identifiers for easier comparison
+        let untagged_version = semver::Version::new(
+            current_version.major,
+            current_version.minor,
+            current_version.patch,
+        );
+
+        let mut matching_version = true;
```

Comment:
I would suggest moving the logic out of the filter closure instead, with something among the lines of `let matching_version = packages.iter().all(|pkg| { ... });`